tests/encryption_test.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/shutter-network/shutter/shlib/shcrypto"
+	blst "github.com/supranational/blst/bindings/go"
+	"gotest.tools/assert"
+)
+
+func bigToScalar(i *big.Int) *blst.Scalar {
+	max256Bit := new(big.Int).Lsh(big.NewInt(1), 256)
+	normalized := new(big.Int).Mod(i, max256Bit)
+
+	b := make([]byte, 32)
+	normalized.FillBytes(b)
+	s := new(blst.Scalar)
+	s.FromBEndian(b)
+	return s
+}
+
+func generateP2(i *big.Int) *blst.P2Affine {
+	s := bigToScalar(i)
+	return blst.P2Generator().Mult(s).ToAffine()
+}
+
+func (s *TestShutterService) makeKeys() (*shcrypto.EonPublicKey, *shcrypto.EpochSecretKey, *shcrypto.EpochID) {
+	s.T().Helper()
+	n := 3
+	threshold := uint64(2)
+	epochID := shcrypto.ComputeEpochID([]byte("epoch1"))
+
+	ps := []*shcrypto.Polynomial{}
+	gammas := []*shcrypto.Gammas{}
+	for i := 0; i < n; i++ {
+		p, err := shcrypto.RandomPolynomial(cryptorand.Reader, threshold-1)
+		assert.NilError(s.T(), err)
+		ps = append(ps, p)
+		gammas = append(gammas, p.Gammas())
+	}
+
+	eonSecretKeyShares := []*shcrypto.EonSecretKeyShare{}
+	epochSecretKeyShares := []*shcrypto.EpochSecretKeyShare{}
+	eonSecretKey := big.NewInt(0)
+	for i := 0; i < n; i++ {
+		eonSecretKey.Add(eonSecretKey, ps[i].Eval(big.NewInt(0)))
+
+		ss := []*big.Int{}
+		for j := 0; j < n; j++ {
+			s := ps[j].EvalForKeyper(i)
+			ss = append(ss, s)
+		}
+		eonSecretKeyShares = append(eonSecretKeyShares, shcrypto.ComputeEonSecretKeyShare(ss))
+		_ = shcrypto.ComputeEonPublicKeyShare(i, gammas)
+		epochSecretKeyShares = append(epochSecretKeyShares, shcrypto.ComputeEpochSecretKeyShare(eonSecretKeyShares[i], epochID))
+	}
+	eonPublicKey := shcrypto.ComputeEonPublicKey(gammas)
+	eonPublicKeyExp := (*shcrypto.EonPublicKey)(generateP2(eonSecretKey))
+	assert.Assert(s.T(), eonPublicKey.Equal(eonPublicKeyExp))
+	epochSecretKey, err := shcrypto.ComputeEpochSecretKey(
+		[]int{0, 1},
+		[]*shcrypto.EpochSecretKeyShare{epochSecretKeyShares[0], epochSecretKeyShares[1]},
+		threshold)
+	assert.NilError(s.T(), err)
+
+	return eonPublicKey, epochSecretKey, epochID
+}
+
+func (s *TestShutterService) TestGetDataForEncryption() {
+	ctx := context.Background()
+	sender, err := generateRandomETHAddress()
+	s.Require().NoError(err)
+	identityPrefix, err := generateRandomBytes(32)
+	s.Require().NoError(err)
+	identityPrefixStringified := hex.EncodeToString(identityPrefix)
+	blockNumber := rand.Uint64()
+
+	eon := rand.Uint64()
+
+	eonPublicKey, _, _ := s.makeKeys()
+
+	s.ethClient.
+		On("BlockNumber", ctx).
+		Return(blockNumber, nil).
+		Once()
+
+	s.keyperSetManagerContract.
+		On("GetKeyperSetIndexByBlock", nil, blockNumber).
+		Return(eon, nil).
+		Once()
+
+	s.keyBroadcastContract.
+		On("GetEonKey", nil, eon).
+		Return(eonPublicKey.Marshal(), nil).
+		Once()
+
+	data, err := s.cryptoUsecase.GetDataForEncryption(ctx, sender, identityPrefixStringified)
+	s.Require().Nil(err)
+
+	identity := common.ComputeIdentity(identityPrefix, ethCommon.HexToAddress(sender))
+
+	s.Require().Equal(data.Eon, eon)
```

Comment:
In the responses we send back to the user, we sometimes prefix some fields with the hex prefix 0x (when sending the decryption key for example), sometimes not, I think we should have all the hex values prefixed for consistency, you could add a helper function that properly formats the response fields in hexadecimal format?